raft-tech/TANF-app
# 3014 - Outdated submissions banner

## Summary of Changes
Pull request closes #3014 
* ~~adds outdated submissions banner for files submitted before 5/31/2024~~
* ~~removes the error report link for those files (replaces with "Not Available")~~
+ combine "submitted by" and "submitted on"

## How to Test
```
cd tdrs-backend && docker compose up
cd tdrs-frontend && docker compose up --build
```

1. Open http://localhost:3000/ and sign in.
1. Submit a data file. Make sure the banner does not appear, and the error reports download link is shown.
1. Modify the `created_at` field of the datafile to before 5/31/2024
   ```bash
   docker compose exec web python manage.py shell_plus

   f = DataFile.objects.all().first()
   from django.utils import timezone
   from datetime import datetime
   f.created_at = datetime(2023, 10, 11)
   f.save()
   ```
1. Return to the datafiles page. Ensure the banner displays and the error report download link is replaced with "Not Available"

## Deliverables
_More details on how deliverables herein are assessed included [here](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverables)._

### [Deliverable 1: Accepted Features](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

Checklist of ACs:
~~+ [ ] Banner appears on submission history if 1 or more files are present for viewed quarter that were submitted prior to May 31, 2024~~
~~+ [ ] Banner does not appear if a given quarter doesn't have data submitted before May 31 2024~~
~~+ [ ] Language matches spec "Please note that error reports and submission history content for files submitted prior to May 31, 2024 may be outdated. Please resubmit to get access to updated information."~~
+ [x] combine "submitted by" and "submitted on"

### [Deliverable 2: Tested Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
+ Are code coverage minimums met?

### [Deliverable 3: Properly Styled Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)


### [Deliverable 4: Accessible](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)



### [Deliverable 5: Deployed](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)


### [Deliverable 6: Documented](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)


### [Deliverable 7: Secure](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)


### [Deliverable 8: User Research](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

Research product(s) clearly articulate(s):
